### PR TITLE
created app user to implement login/registartion

### PR DESCRIPTION
**Pull request by Shyamaskamath**
created an app user to implement  login and registration

### CODE REVIEW

File: user/forms.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from django.contrib.auth.models import User
+''' cform for new user creation '''
+class UserRegistrationForm(UserCreationForm):
+    email = forms.EmailField()
+    class Meta:
+        model = User
+        fields = ['username','email','password1','password2']
+        
+''' Form for user login '''
+class LoginForm(forms.Form):
```
Comment (zudheer) [inline user/forms.py]: docstrings are in the wrong format. 
It should be 
class LoginForm(forms.Form):
    """Form for user login."""

File: user/urls.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+urlpatterns = [
+   
+    path('register/',views.RegisterationFormView.as_view(),name="signup"),
+    path('home/',views.HomePageView.as_view(),name="home"),
+    path('login/',views.LoginView.as_view(),name="login"),
+    path('logout/',auth_views.LogoutView.as_view(next_page="login"), name='logout'),
+   
+
```
Comment (zudheer) [inline user/urls.py]: remove unwanted spaces

File: user/views.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+class RegisterationFormView(generic.CreateView):
+    template_name = 'user/register.html'
+    form_class = UserRegistrationForm
+    success_url = reverse_lazy("login")
+
+'''Function based view for new user creation/registration '''
+
+def registerformview(request):
+    if request.method =='POST':
```
Comment (zudheer) [inline user/views.py]: if this function is not used, remove it. else convert this to class based view

File: user/views.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            form = UserRegistrationForm()
+            return redirect('login')
+    else:
+        form = UserRegistrationForm()
+    return render(request,'user/register.html',{'form':form})
+
+'''function based view to implement Login'''
+
+# def loginformview(request):
+#     if request.method == 'POST':
```
Comment (zudheer) [inline user/views.py]: remove the commented lines

File: user/views.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+#                 login(request, user)
+#                 return redirect('home')
+
+#     form = LoginForm()
+#     return render(request = request,
+#                     template_name = "user/login.html", context={"form":form})
+
+'''class based view  to implement login '''
+class LoginFormView(generic.View):
+    template_name = 'user/login.html'
```
Comment (zudheer) [inline user/views.py]: update all docstrings in correct format

**Comment by Shyamaskamath** [comment]
created an app 'user' to implement login/registration

File: user/views.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    form_class = LoginForm
+
+    def get(self,request):
+        form = self.form_class
```
Comment (zudheer) [inline user/views.py]: no need of  defining  get unless you are overriding anything

File: user/views.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        return render(request,self.template_name,{"form":form})
+
+    
+    def post(self,request):
+            form = self.form_class(request.POST)
+            if form.is_valid():
```
Comment (zudheer) [inline user/views.py]: use def form_valid() and form_invalid() instead of directly modifying the post

**Pull request closed without merging**